html/controls/plt/plt-input.js

```diff
@@ -600,4 +623,113 @@ function preparePenaltyEditor() {
 		}
 	}
 }
+
+var annotationEditor;
+
+function openAnnotationEditor(teamId, skaterId) {
+	var prefix = 'ScoreBoard.Team('+teamId+').Skater('+skaterId+').';
+	var skaterNumber = WS.state[prefix + 'Number'];
+	var position = WS.state[prefix + 'Position'].slice(2);
+	var annotationPath = ').TeamJam('+teamId+').Fielding(' + position + ').Annotation';
+	if (isTrue(WS.state['ScoreBoard.InJam'])) {
+		annotationPath = 'ScoreBoard.Period(' + WS.state['ScoreBoard.CurrentPeriodNumber'] +
+		').Jam(' + WS.state['ScoreBoard.Period('+WS.state['ScoreBoard.CurrentPeriodNumber']+').CurrentJamNumber']
+		+ annotationPath;
+	} else {
+		annotationPath = 'ScoreBoard.Jam(' + WS.state['ScoreBoard.UpcomingJamNumber'] + annotationPath;
+	}
+	annotationEditor.data('skaterNumber', skaterNumber);
+	annotationEditor.data('position', position);
+	annotationEditor.find('#subDropdown').val(skaterId);
+	annotationEditor.find('#annotation').val(WS.state[annotationPath]);
+	annotationEditor.find('.Box').toggleClass('Hide', !isTrue(WS.state[prefix + 'PenaltyBox']));
+	annotationEditor.dialog('open');
+}
+
+function prepareAnnotationEditor(teamId) {
+	$(initialize);
+	
+	var subDropdown = $('<select>').attr('id', 'subDropdown');
+	var annotationField = $('<input type="text">').attr('size', '35').attr('id', 'annotation');
+	var jamPrefix;
+	
+	function initialize() {
+		var table = $('<table>').appendTo($('#AnnotationEditor'));
+		var row = $('<tr>').addClass('Box').appendTo(table);
+		$('<td>').append(subDropdown)
+				.append($('<button>').text('Substitute').button().click(function() {
+					var prefix = jamPrefix + annotationEditor.data('position') + ').';
+					WS.Set(prefix + 'Skater', subDropdown.val());
+					WS.Set(prefix + 'Annotation', 'Substitute for #' + annotationEditor.data('skaterNumber'));
+					annotationEditor.dialog('close');
+				})).appendTo(row);
+		$('<td>').append($('<button>').text('Self Report').button().click(function() { leaveBox('Self Report');})).appendTo(row);
```

Comment:
No penalty and Penalty overturned are what the statsbook manual has

(and Left early to be complete)

Reply:
Those are only defined for PBT. But we might as well use them here.